web/interceptor/access_control.go

```diff
@@ -0,0 +1,150 @@
+package interceptor
+
+import (
+	"context"
+	"strings"
+
+	"github.com/quickfeed/quickfeed/qf"
+	"github.com/quickfeed/quickfeed/web/auth"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+type (
+	role      int
+	roles     []role
+	requestID interface {
+		FetchID(string) uint64
+	}
+)
+
+const (
+	// user role implies that user attempts to access information about himself.
+	user role = iota
+	// group role implies that the user is a course student + a member of the given group.
+	group
+	// student role implies that the user is enrolled in the course with any role.
+	student
+	// teacher: user enrolled in the course with teacher status.
+	teacher
+	// courseAdmin: an admin user who is also enrolled into the course.
+	courseAdmin
+	// admin is the user with admin privileges.
+	admin
+)
+
+// If there are several roles that can call a method, a role with the least privilege must come first.
+// If method is not in the map, there is no restrictions to call it.
+var access = map[string]roles{
+	"GetUser":                {},
```

Comment:
What does it mean when the `roles` are empty? Should we require to specify `everyone` or something, instead of leaving it empty?